### ISSUE TITLE
Question: sdkman_rosetta 2_compatible doesn't work

### ISSUE THREAD

**Issue by polev-evgeny** [sdkman/sdkman-cli#1238]
Hi. I try to use x86_64 on my mac m1

I have installed sdkman
SDKMAN!
script: 5.18.2
native: 0.4.2

I have switched flag sdkman_rosetta 2_compatible=true, but I still get arm versions when I use command `sdk list java`.
I see it in header - Available Java Versions for macOS ARM 64bit
I tried to look for sdkman_rosetta 2_compatible flag usage, but I don't found it.
Could you help me ?

**Comment by marc0der** [comment]
Hi @polev-evgeny, did you try opening a new terminal after making the change? The config file is sourced and all entries show up as environment variables. Can you confirm with `echo` that it is being set correctly In your shell?

**Comment by polev-evgeny** [comment]
@marc0der Yes, I did it. Also I tried to restart mac but I still got the same versions.
I made investigation and I found place where you detect what os should be used.
I changed variable platform manually and it works
I think that question can be closed but maybe you will made investigation because I think somewhere in script some mistake to variables order

**Comment by kyuds** [comment]
@polev-evgeny could you also tell me where you found the place to detect the OS type being used? Thank you in advance!

**Comment by caelcs** [comment]
@kyuds you need to edit .sdkman/var/platform and change to **darwinarm64**

**Comment by marc0der** [comment]
@polev-evgeny we haven't had any other reports of platform inference failing, and we have _many_ Mac M1 users. I suspect this isn't a bug but rather a local setup issue on your machine. Otherwise, could you point me to where you think this bug is?

**Comment by cinyearchan** [comment]
same issue

set ~/.sdkman/etc/config sdkman_rosetta2_compatible=true
exit all iterm2
start iterm.app with rosetta
type "arch" and got "i386"
~/.sdkman/var/platform remains "darwinarm64"

sdk list java -> got arm64 list
-------------------------------
when I set ~/.sdkman/var/platform "darwinx64"
sdk list java -> got x64  list

**Comment by marc0der** [comment]
> start iterm.app with rosetta

@cinyearchan, I'm not sure I understand what you mean by this. Can you explain? We infer the platform during SDKMAN installation, which is then written to the file at `~/.sdkman/var/platform`. This file will not be updated again unless you manually change it. 

I also want to clarify the current situation with Rosetta 2. Rosetta 2 was bridging technology provided by Apple so that x86 compiled binaries could run on ARM hardware. It was much needed at the time as most vendors weren't compiling binaries for the ARM architecture. Fast forward many years, and we now have ARM-compiled binaries for _all_ JDKs. It no longer makes sense to use Rosetta 2 because the JDKs compiled for ARM are far more performant than x64 binaries running under Rosetta 2. I'd recommend sticking to the default ARM binaries unless you have a compelling reason not to.

Finally, the complete list of supported configurations can be found [here](https://sdkman.io/usage#configuration) on our website.
